### PR TITLE
Add nameless_api.conf to repo.

### PR DESCRIPTION
**Pull request by Jason-Doze**
Add api_deploy.sh to repo.

Add ssh command in pi local to validate the server is running.

Add env variable command to run pi_local.sh.

### CODE REVIEW

File: api_deploy.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  sudo apt install -y nodejs 
+fi
+
+# Install NPM and its dependencies
+if [ -d node_modules ] 
+then
+  echo -e "\n==== Node_modules installed ====\n"
+else 
+  echo -e "\n==== Installing node_modules ====\n"
+  cd /home/jasondoze/expressapi && npm install
```
Comment (stephenlauck) [inline api_deploy.sh:39]: check out the bash command "pushd"

https://linuxhint.com/bash_pushd_command/

File: nameless_api.conf [hunk 2]
```diff
@@ -2,9 +2,10 @@
...
-  # Set the server name for this block to "nameless.api"
-  server_name nameless.api;
-  # Define a new location block for the root path ("/")
+  # Set the server name for this block to relativepath
+  server_name relativepath;
+
+  # Define a location block for the static page app at the root path ("/")
   location / {
     # Forward incoming requests to a server running on localhost:3000
     proxy_pass http://localhost:3000;
```
Comment (stephenlauck) [inline nameless_api.conf:11]: So you're serving the static content from node. What if you wanted to server the static content from nginx?

**Review by stephenlauck — CHANGES_REQUESTED** [review]

File: nameless_api.conf (hunk 2 above)
Comment (Jason-Doze) [reply nameless_api.conf:11]: @stephenlauck  
This is the config for the api app. I have a separate relativepath.conf for the static site, not sure if thats the protocol. Which is preferred, one .conf for both locations or separate .conf files for each location? I was thinking the latter.

Comment (stephenlauck) [reply nameless_api.conf:11]: There are two places you can serve static content: 1. from your upstream node app 2. from nginx directly. If you were to serve from nginx directly, you can put your static content into `/var/www/relativepath` and then point the location to that path. This means that nginx isn't even going upstream for the static but is serving it directly. Let me know if this makes sense.

Comment (Jason-Doze) [reply nameless_api.conf:11]: I remember serving the viking app at /var/www/viking with nginx, that makes sense. 

The reverse proxy and upstream etc are confusing. I'll search for a visual of these so I can better wrap my head around it.

**Pull request merged**